lib/ohai/plugins/joyent.rb

```diff
@@ -25,24 +25,11 @@
   depends 'os', 'platform', 'virtualization'
 
   def collect_product_file
-    lines = []
-    if ::File.exists?("/etc/product")
-      ::File.open("/etc/product") do |file|
-        while line = file.gets
-          lines << line
-        end
-      end
-    end
-    lines
+    ::File.read("/etc/product").strip.split("\n") rescue []
   end
 
   def collect_pkgsrc
-    if File.exist?('/opt/local/etc/pkg_install.conf')
-      sm_pkgsrc = ::File.read("/opt/local/etc/pkg_install.conf").split("=")
-      sm_pkgsrc[1].chomp
-    else
-      nil
-    end
+    ::File.read("/opt/local/etc/pkg_install.conf").strip.split("=", 2).last rescue nil
```

Comment:
Can we make this rescue more specific? The way it's written now we'll rescue anything in `.strip.split("=", 2).last` when really we just want to rescue `ENOENT` and `EACCES` on the file read.